### PR TITLE
Update Hooks documentation

### PR DESCRIPTION
**Pull request by aurelien-reeves**
# Description

Update the cucumber hooks documentation to consider new `BeforeAll`/`AfterAll` hooks, the new `InstallPlugin` ruby hook, and the deprecation of `AfterConfiguration`.

# Motivation & context

There have been changes recently regarding hooks.

Refs. https://github.com/cucumber/cucumber-ruby/pull/1570

## Type of change

- New feature (non-breaking change which adds new behaviour)
- Breaking change (will cause existing functionality to not
  work as expected)

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
Note: cucumber-ruby with the changes to the hooks has not been released yet, so wait cucumber@7.1.0 before merging this.

File: content/docs/cucumber/api.md
Comment (mpkorstanje) [inline content/docs/cucumber/api.md:609]: Not yet available in Cucumber JVM v6. Will be in V7.

**Review by davidjgoss — COMMENTED** [review]
For JS, I would probably just give one example and make it in the `async function` syntax now as that will be the dominant one. WDYT?

File: content/docs/cucumber/api.md
Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: I though v7 had been released 😅

But actually you have spotted an issue here: the documentation do not care about versions of cucumber. Someone with an older version of cucumber will ever find the appropriate documentation.

Comment (mpkorstanje) [reply content/docs/cucumber/api.md:609]: That's always been like that and usually not a big problem.

Comment (mpkorstanje) [reply content/docs/cucumber/api.md:609]: I did want to release v7. Just haven't gotten around to it yet. :laughing:

Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: So, we'll see at the moment of merging that PR, if v7 not yet released, we'll remove the java part, replacing it with "coming soon"

**Comment by aurelien-reeves** [comment]
> 
> 
> For JS, I would probably just give one example and make it in the `async function` syntax now as that will be the dominant one. WDYT?

I've updated the API documentation to replace all the javascript examples with a single one with an `async function()`. Does it looks good?

File: content/docs/cucumber/api.md
Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: What about simply redirecting such technical documentation to the repositories like [cucumber-js](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/hooks.md), and [cucumber-ruby](https://github.com/cucumber/cucumber-ruby/blob/main/features/docs/writing_support_code/hooks/README.md)?

Those remain tech documentations: there is no need to keep those away from the code as opposed to gherkin documentation and references, guides, and other content on cucumber.io/docs

It will be versioned: users will have the possibility to find the appropriate documentation to their versions

It will be easier for us, as developers, to keep it up-to-date and to make it part of our development processes (I mean, like we keep the changelog up-to-date as part of our PR and reviewers take care of this, we could do the same with such tech documentations)

What do you think?

**Comment by davidjgoss** [comment]
@aurelien-reeves yep LGTM on the JavaScript side, thanks!

File: content/docs/cucumber/api.md
Comment (davidjgoss) [reply content/docs/cucumber/api.md:609]: For my part I think the technical docs on this site should be super thin (closer to concepts than specific APIs) and hand off to relevant docs in the implementation repos. That might just be me though...

Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: yes, I have the same though

File: content/docs/cucumber/api.md
Comment (luke-hill) [inline content/docs/cucumber/api.md]: AfterAll

File: content/docs/cucumber/api.md
Comment (luke-hill) [inline content/docs/cucumber/api.md]: extra line

File: content/docs/cucumber/api.md
Comment (luke-hill) [inline content/docs/cucumber/api.md]: does not support the `InstallPlugin` hook.

**Review by luke-hill — COMMENTED** [review]
Minor language tweaks, all seems good though

File: content/docs/cucumber/api.md
Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: @mpkorstanje should I revert the changes for Java despite the release of cucumber-jvm 7 RC1?

File: content/docs/cucumber/api.md
Comment (olleolleolle) [inline content/docs/cucumber/api.md]: Minor: Cucumber.js is the name of the project.

```suggestion
{{% text "javascript" %}}Cucumber.js does not support the `InstallPlugin` hook.{{% /text %}}
```

**Review by luke-hill — APPROVED** [review]

**Review by davidjgoss — APPROVED** [review]

File: content/docs/cucumber/api.md
Comment (mpkorstanje) [reply content/docs/cucumber/api.md:609]: Probably best for now. Maybe split them out in to a separate pull request.

Comment (aurelien-reeves) [reply content/docs/cucumber/api.md:609]: 👍 
Will do before merging if cucumber-jvm 7 still not released.

Won't merge this now as cucumber-ruby still need to be released too. But I think we may release it pretty soon.

**Pull request merged**
